store/src/main/java/com/dropbox/android/external/store4/impl/RealStore.kt

```diff
@@ -78,31 +80,38 @@ internal class RealStore<Key : Any, Input : Any, Output : Any>(
         sourceOfTruth = this.sourceOfTruth
     )
 
-    override fun stream(request: StoreRequest<Key>): Flow<StoreResponse<Output>> {
-        return if (sourceOfTruth == null) {
-            @Suppress("UNCHECKED_CAST")
-            createNetworkFlow(
-                request = request,
-                networkLock = null
-            ) as Flow<StoreResponse<Output>> // when no source of truth Input == Output
-        } else {
-            diskNetworkCombined(request, sourceOfTruth)
+    override fun stream(request: StoreRequest<Key>): Flow<StoreResponse<Output>> =
+        flow<StoreResponse<Output>> {
+            val cached = memCache?.get(request.key)
```

Comment:
we shouldn't read the cache unless cache is requested.
if we change this to:

val cached = if (readMemry) memCache?.get(request.key) else null

then we can remove the check from piggybackOnly and also avoid reading cache unless we need to.

also, feels like this logic should just move into the SOT false branch. If there is a source of truth, we don't need this as far as i can see.